### ISSUE TITLE
quotes does not catch errors in strings wrapped in {...} within jsx

### ISSUE THREAD

**Issue by deniseli** [eslint/eslint#7084]
**What version are you using?**
3.4.0

**What did you do?**
Run lint with `quotes` rule turned on (prefer single quotes) over jsx code that looks like:
`<SomeComponent someProp={"blah"} />`

**What happened?**
The linter passed, when it should have errored on the double quotes. This is not covered by jsx-quotes because the string is wrapped in `{...}`, which makes it a JS string and not a jsx string.

**What did you expect to happen?**
Error for violating the `quotes` rule

**Comment by ilyavolodin** [comment]
Verified with the following code:

```js
/*eslint quotes: ["error", "single"] */
/*eslint jsx-quotes: ["error", "prefer-single"]*/

function a() {
	return <div blah={"blah"} />;
}
```

**Comment by mysticatea** [comment]
I'm working on this.
